tests/components/mqtt_statestream/test_init.py

```diff
@@ -3,375 +3,354 @@
-        self.hass.block_till_done()
-
-        # Make sure 'on' was published to pub/fake/entity/state
-        mock_pub.assert_called_with(self.hass, "pub/fake/entity/state", "on", 1, True)
-        assert mock_pub.called
-
-        mock_pub.reset_mock()
-        # Set a state of an entity that shouldn't be included
-        mock_state_change_event(self.hass, State("fake.entity2", "on"))
-        self.hass.block_till_done()
-
-        assert not mock_pub.called
+async def add_statestream(
+    hass,
+    base_topic=None,
+    publish_attributes=None,
+    publish_timestamps=None,
+    publish_include=None,
+    publish_exclude=None,
+):
+    """Add a mqtt_statestream component."""
+    config = {}
+    if base_topic:
+        config["base_topic"] = base_topic
+    if publish_attributes:
+        config["publish_attributes"] = publish_attributes
+    if publish_timestamps:
+        config["publish_timestamps"] = publish_timestamps
+    if publish_include:
+        config["include"] = publish_include
+    if publish_exclude:
+        config["exclude"] = publish_exclude
+    return await async_setup_component(
+        hass, statestream.DOMAIN, {statestream.DOMAIN: config}
+    )
+
+
+async def test_fails_with_no_base(hass, mqtt_mock):
+    """Setup should fail if no base_topic is set."""
+    assert await add_statestream(hass) is False
+
+
+async def test_setup_succeeds_without_attributes(hass, mqtt_mock):
+    """Test the success of the setup with a valid base_topic."""
+    assert await add_statestream(hass, base_topic="pub")
+
+
+async def test_setup_succeeds_with_attributes(hass, mqtt_mock):
+    """Test setup with a valid base_topic and publish_attributes."""
+    assert await add_statestream(hass, base_topic="pub", publish_attributes=True)
+
+
+async def test_state_changed_event_sends_message(hass, mqtt_mock):
+    """Test the sending of a new message if event changed."""
+    e_id = "fake.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(hass, base_topic=base_topic)
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_statestream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    # Set a state of an entity
+    mock_state_change_event(hass, State(e_id, "on"))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Make sure 'on' was published to pub/fake/entity/state
+    mqtt_mock.async_publish.assert_called_with("pub/fake/entity/state", "on", 1, True)
+    assert mqtt_mock.async_publish.called
+
+
+async def test_state_changed_event_sends_message_and_timestamp(hass, mqtt_mock):
+    """Test the sending of a message and timestamps if event changed."""
+    e_id = "another.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)